Documents/Technical Specifications/Technical.md

```diff
@@ -0,0 +1,494 @@
+While the cube is moving, any other interractions will be disabled to avoid unwanted behavior.
+
+### Obstacle
+
+Obstacles will be placed in the level to prevent the player from reaching the cup of coffee.
+Obstacles will be items integrated in the environment and might take any forms ( Wall, plate, etc. ).
+Each obstacle will have a collision box for the player to not glitch through it.
+Each obstacle will have a material to make it visible.
+Each obstacles placement and details will be define in the [Level section](#level).
+
+### Sticky
+
+Sticky will be a mechanic that will allow the player to stick to walls.
+It will be shown as a green particle effect on the wall as if the wall was dirty.
+It will activate when the player enter a special zone.
+And will stay activated as long as the player stay in that zone and follow the predefined path.
+The controls stay relative to the camera.
+It means old controls become as follow:
+
+- Forward: Z -> Forward
+- Backward: S -> Backward
+- Left: Q -> Down
+- Right: D -> Up
+  
+If the player is sticked to a wall and decide to leave the predefined path, the sticky mechanic will be deactivated and the player will fall resulting in a death.
+
+### Catapult
+
+Catapult will be a mechanic that will allow the player to launch itself in the air.
+It will be represented as a spoon.
+It will activate when the player enter the back of the spoon.
+It will launch the player in the direction of the spoon from point A to point B.
+
+There is no laucnh control, air control or gravity control.
+The controls will be desactivated while the player is in the air.
+The catapult have an animation.
+
+### Teleport
+
+Teleport will be a mechanic that will allow the player to teleport to a specific location.
+It will be represented as mouse hole.
+It will activate when the player enter the mouse hole.
+It will teleport the player to the other side of the mouse hole.
+The teleport will be instant.
+It will be one sided ( No come back ).
+It have an animation.
+
+### Death
+
+The player have multiple ways to die.
+
+#### Death by falling
+
+The player will die if he fall from a height.
+For a better gameplay, the player will be teleported to the start of the level when he fall from a height.
+
+#### Death by water
+
+The player will die if he fall in the water.
+The player will be able to stay for a maximum of 3.00 seconds in the water before dying.
+This obstacle will be represented as a water puddle.
+To have the most realistic gameplay, the water will have to be placed on a straight surface.
+To keep the gameplay fluid, the player will be teleported to the start of the level when he fall in the water for more than 3.00 sec.
+
+#### Death by heat
+
+The player will die if he stay in the heat for more than 5.00 sec.
+This obstacle will be represented as a light.
+To keep the gameplay fluid, the player will be teleported to the start of the level when he stay in the heat for more than 5.00 sec.
+
```

Comment:
We should add effects to notice the player that he's actually dying.

Reply:
Added in the HUD section